### ISSUE TITLE
REST API order info -> "parent" products sku inconsistency with product_id 

### ISSUE THREAD

**Issue by tsoftp** [magento/magento2#21231]
### Preconditions (*)

1.Magento 2.4-develop
2.Using Luma test sample data 
3.Using Postman for REST API calls.

### Steps to reproduce (*)

1.Simply create an order with a configurable product (i.e. a shirt size M color grey or something like that)
2.Make a REST API call to return that order/list of orders.

### Expected result (*)

1. Receive the API response for the order.
2. The product I ordered is listed in the "Items" array and its "parent_item" attribute shows the original configurable product's data.

### Actual result (*)

1. The product I ordered is listed in the "Items" array and its "parent_item" attribute shows the child's "sku" attribute instead of the sku of the parent_item.

In other words it says that the parent's Sku is the same as the simple product sku which is simply not true, despite it having the correct "name" and "item_id" attribute values.

2. Example info attached with what I am describing happening. File has comments on the lines I'm talking about.
[bug.txt](https://github.com/magento/magento2/files/2865767/bug.txt)

Thank you!

**Comment by magento-engcom-team** [comment]
Hi @tsoftp. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@tsoftp do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by tsoftp** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @tsoftp. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @tsoftp, here is your [Magento instance](https://i-21231-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21231-2-3-develop.instances.magento-community.engineering/admin](https://i-21231-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by tsoftp** [comment]
> Hi @tsoftp. Thank you for your report.
> To help us process this issue please make sure that you provided the following information:
> 
> * [ ]  Summary of the issue
> * [ ]  Information on your environment
> * [ ]  Steps to reproduce
> * [ ]  Expected and actual results
> 
> Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue:
> 
> `@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release
> 
> For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.
> 
> @tsoftp do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
> 
> * [X ]  yes
> * [ ]  no

Just reproduced it.

**Comment by magento-engcom-team** [comment]
Hi @tsoftp. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @tsoftp, here is your [Magento instance](https://i-21231-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-21231-2-3-develop.instances.magento-community.engineering/admin](https://i-21231-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by tsoftp** [comment]
I just reproduced it on the first instance @magento-engcom-team created. (2nd instance I created by accident by quoting him/it)

> @tsoftp do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
> 
> * [x]  yes
> * [ ]  no

**Comment by ryandiamond23** [comment]
@magento-engcom-team give me 2.2.7

**Comment by ryandiamond23** [comment]
@magento-engcom-team give me 2.2.7 instance

**Comment by magento-engcom-team** [comment]
Hi @ryandiamond23. Thank you for your request. I'm working on Magento 2.2.7 instance for you

**Comment by magento-engcom-team** [comment]
Hi @ryandiamond23, here is your [Magento instance](https://i-21231-2-2-7.instances.magento-community.engineering).
Admin access: [https://i-21231-2-2-7.instances.magento-community.engineering/admin](https://i-21231-2-2-7.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98287`, `MAGETWO-98288` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @hiren0241. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if you want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by diamondavocado** [comment]
How can this **still** be an issue after 3 years? This is **super basic** stuff and a severe bug which makes the API data totally unreliable.

**Comment by engcom-Delta** [comment]
Hi @tsoftp ,

Thanks for your reporting and collaboration.
We have verified the issue in latest 2.4-develop instance and the issue is reproducible.
Kindly refer the screenshots.

**Steps to reproduce**
1.Simply create an order with a configurable product (i.e. a shirt size M color grey or something like that)
2.Make a REST API call to return that order/list of orders.
Response - 
![image](https://github.com/user-attachments/assets/c9e702d1-8e03-4488-82ff-09ef8616c518)
[api response.txt](https://github.com/user-attachments/files/17130852/api.response.txt)

Thanks.

**Comment by diamondavocado** [comment]
Great job guys, only took 5 and a half years to confirm the existence of this bug.

Meanwhile anyone with a room temperature IQ has long since stopped using Magento and started using something better such as literally anything else.
